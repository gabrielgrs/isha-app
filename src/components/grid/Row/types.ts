import { ReactNode } from 'react'

export type Row = {
  children: ReactNode
  id?: string
  marginBottom?: string | number
  justifyItems?: 'start' | 'center' | 'end'
  alignItems?: 'start' | 'center' | 'end'
  rowGap?: string
}
