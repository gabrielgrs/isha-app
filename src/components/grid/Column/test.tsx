import { render, screen } from 'utils/test'

import Column from './index'

describe('<Column />', () => {
  it('should render Column', () => {
    const { container } = render(<Column size={12}>I am a Column</Column>)

    expect(screen.queryByText('I am a Column')).toHaveStyle({ gridColumn: 'span 12' })
    expect(container).toMatchSnapshot()
  })
})
