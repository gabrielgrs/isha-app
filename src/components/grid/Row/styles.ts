import styled from 'styled-components'

import * as T from './types'

export const Wrapper = styled.div<T.Row>`
  display: grid;
  column-gap: 20px;
  row-gap: ${({ rowGap = '32px' }) => rowGap};
  grid-template-columns: repeat(12, 1fr);
  margin-bottom: ${({ marginBottom }) => marginBottom};
  justify-items: ${({ justifyItems }) => justifyItems};
  align-items: ${({ alignItems }) => alignItems};
`
