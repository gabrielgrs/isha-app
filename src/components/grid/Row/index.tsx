import * as S from './styles'
import * as T from './types'

function Row({ children, ...rest }: T.Row) {
  return <S.Wrapper {...rest}>{children}</S.Wrapper>
}

export default Row
