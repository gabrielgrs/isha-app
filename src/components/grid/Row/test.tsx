import { render } from 'utils/test'

import Row from './index'

describe('<Row />', () => {
  it('should render Row', () => {
    const { container } = render(<Row>I am a Row</Row>)

    expect(container).toMatchSnapshot()
  })
})
