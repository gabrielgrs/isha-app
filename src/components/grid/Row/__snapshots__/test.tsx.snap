// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> should render Row 1`] = `
.c0 {
  display: grid;
  -webkit-column-gap: 20px;
  column-gap: 20px;
  row-gap: 32px;
  grid-template-columns: repeat(12,1fr);
}

<div>
  <div
    class="c0"
  >
    I am a Row
  </div>
</div>
`;
